Examples/Algorithms/TrackFitting/include/ActsExamples/TrackFitting/TrackFittingAlgorithm.hpp

```diff
@@ -52,27 +50,14 @@ class TrackFittingAlgorithm final : public BareAlgorithm {
 
   /// Fit function that takes the above parameters and runs a fit
   /// @note This is separated into a virtual interface to keep compilation units
-  /// small
+  /// small.
   class TrackFitterFunction {
    public:
     virtual ~TrackFitterFunction() = default;
     virtual TrackFitterResult operator()(
         const std::vector<std::reference_wrapper<const IndexSourceLink>>&,
         const TrackParameters&, const GeneralFitterOptions&,
-        std::shared_ptr<Acts::VectorMultiTrajectory>& trajectory) const = 0;
-  };
-
-  /// Fit function that takes the above parameters plus a sorted surface
-  /// sequence for the DirectNavigator to follow
-  /// @note This is separated into a virtual interface to keep compilation units
-  /// small
-  class DirectedTrackFitterFunction {
-   public:
-    virtual ~DirectedTrackFitterFunction() = default;
-    virtual TrackFitterResult operator()(
-        const std::vector<std::reference_wrapper<const IndexSourceLink>>&,
-        const TrackParameters&, const GeneralFitterOptions&,
-        const std::vector<const Acts::Surface*>&,
+        const std::optional<std::vector<const Acts::Surface*>>&,
```

Comment:
This could also be a separate overload, instead of an `optional` argument. Might be more obvious?

Reply:
Yeah indeed, might be better!